src/dbxdeploy/repos/RepoManager.py

```diff
@@ -19,29 +20,41 @@ def __init__(
         self.__repo_root_dir = repo_root_dir
         self.__repo_path = repo_path
 
-    def create_or_update(self, repo_url, branch, tag, repo_name):
+    def create_or_update(self, repo_url, branch, tag, repo_name, force):
         repo_path = self.__repo_path.format(repo_name=repo_name)
-
-        self.__logger.info(f"Trying to update repo at {repo_path}")
+        self.__logger.info(f"Updating repo at {repo_path}")
         self.__workspace_api.mkdirs(self.__repo_root_dir)
 
+        repo_list = self.__repos_api.list(self.__repo_root_dir, None).get("repos", [])
+
         try:
             repo_id = self.__repos_api.get_repo_id(repo_path)
-            self.__check_for_duplicates()
+            self.__check_duplicates(repo_list)
             self.__check_correct_project(repo_id, repo_url)
+            if force:
+                self.__logger.info("Re-cloning repo")
+                self.__repos_api.delete(repo_id)
+                repo_id = self.__repos_api.create(url=repo_url, provider=self.__get_provider_from_url(repo_url), path=repo_path)["id"]
         except RuntimeError:
             self.__logger.info("Repo not found, cloning new repo")
-            self.__check_for_duplicates(with_branch=branch)
+            self.__check_duplicates(repo_list)
+            self.__check_valid_branch(repo_list, branch)
             repo_id = self.__repos_api.create(url=repo_url, provider=self.__get_provider_from_url(repo_url), path=repo_path)["id"]
 
-        self.__repos_api.update(repo_id=repo_id, branch=branch, tag=tag)
+        try:
+            self.__repos_api.update(repo_id=repo_id, branch=branch, tag=tag)
+        except HTTPError as e:
+            if e.response.json().get("error_code") == "GIT_UNKNOWN_REF":
+                self.__logger.info("Error occurred, cleaning up ...")
```

Comment:
When does this happen? Is it safe to "delete everything" when this error occurs?

Reply:
This happens when it's trying to checkout a remote branch that doesn't exist (which would probably be the most common user error if used otside of CI/CD), leaving a repo checked out on master on Databricks. Since the delete would occur only when cloning a new repo, it should be safe.